mmv1/templates/terraform/pre_update/alloydb_cluster.go.erb

```diff
@@ -0,0 +1,67 @@
+// Restrict modification of cluster_type from PRIMARY to SECONDARY as it is an invalid operation
+if d.HasChange("cluster_type") && d.Get("cluster_type") == "SECONDARY" {
```

Comment:
We also want to restrict setting `secondary_config` while `cluster_type` is primary, right? I think we want to check against the new value for each (i.e. `d.Get` rather than HasChange or GetChange), and return an error if `secondary_config` is set and `cluster_type` is primary.

Reply:
Added restriction on setting secondary_config when cluster_type is primary